aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -912,7 +913,7 @@ Tensor mm_batching_rule(const Tensor& self, const Tensor& other) {
   TORCH_INTERNAL_ASSERT(false, "either self or other must be a BatchedTensor");
 }
 
-Tensor cat_batching_rule(TensorList tensors, int64_t dim) {
+Tensor cat_batching_rule(const ITensorList& tensors, int64_t dim) {
```

Comment:
just curious - is there a reason for the switch to const references?

I remember we needed it in `TensorBody.h` to avoid the #include cycles, but does it provide any benefit in the other places where we already take other reference types by value? (`TensorList`, `IntArrayRef`, etc)

Reply:
It looks like `ArrayRef<Tensor>::front()` does bounds checking, while `operator[]` doesn't.

To preserve the existing error behavior, maybe we should add a `front()` method to `IList` that does bounds checking?

Reply:
- We need them here, because `cat` was defined (e.g. `library.def("cat", ...)`) with the `const &`
- I believe they don't provide any benefits in this context

If we want to have non-const value `ITensorList`, I think we could add them when generating the method's signature. i.e. make the argument type for the dispatcher and C++ API for `Tensor[]` as `ITensorList` (no `const &`), and add them (the `const &`) when generating the method's signature in `TensorBody.h`.

Reply:
ohhh yup. If we need them then SGTM